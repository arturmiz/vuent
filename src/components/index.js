import './Vnt.vue';
export { default as VntAutosuggest } from './autosuggest';
export { default as VntButton } from './button';
export { default as VntCheckbox } from './checkbox';
export { default as VntDialog } from './dialog';
export { default as VntHeader } from './header';
export { default as VntIcon } from './icon';
export { default as VntInput } from './input';
export { default as VntLink } from './link';
export { default as VntListView } from './listview';
export { default as VntNavView } from './navview';
export { default as VntPivot } from './pivot';
export { default as VntRadio } from './radio';
export { default as VntRating } from './rating';
export { default as VntSelect } from './select';
export { default as VntSlider } from './slider';
export { default as VntToggle } from './toggle';
