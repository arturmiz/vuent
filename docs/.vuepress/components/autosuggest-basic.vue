<template>
  <div class="playground">
    <vnt-autosuggest v-model="country"
                     :options="countries"></vnt-autosuggest>
    <div class="result">country: {{ country }}</div>
  </div>
</template>

<script>
import { VntAutosuggest } from '../../../src/components';

export default {
  name: 'AutosuggestBasic',
  components: {
    VntAutosuggest
  },
  data() {
    return {
      country: null,
      countries: [
        'Australia',
        'Belgium',
        'Canada',
        'China',
        'France',
        'Germany',
        'Poland',
        'United Kingdom',
        'United States',
      ]
    }
  }
}
</script>
