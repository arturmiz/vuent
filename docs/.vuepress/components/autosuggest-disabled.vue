<template>
  <div class="playground">
    <vnt-autosuggest v-model="country"
                     :options="countries" disabled></vnt-autosuggest>
    <div class="result">country: {{ country }}</div>
  </div>
</template>

<script>
import { VntAutosuggest } from '../../../src/components';

export default {
  name: 'AutosuggestDisabled',
  components: {
    VntAutosuggest
  },
  data() {
    return {
      country: null,
      countries: [
        'Australia',
        'Belgium',
        'Canada',
        'China',
        'France',
        'Germany',
        'Poland',
        'United Kingdom',
        'United States',
      ]
    }
  }
}
</script>
