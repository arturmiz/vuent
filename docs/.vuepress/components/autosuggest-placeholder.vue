<template>
  <div class="playground">
    <vnt-autosuggest v-model="country"
                     :options="countries"
                     placeholder="Type something..."></vnt-autosuggest>
    <div class="result">country: {{ country }}</div>
  </div>
</template>

<script>
import { VntAutosuggest } from '../../../src/components';

export default {
  name: 'AutosuggestPlaceholder',
  components: {
    VntAutosuggest
  },
  data() {
    return {
      country: null,
      countries: [
        'Australia',
        'Belgium',
        'Canada',
        'China',
        'France',
        'Germany',
        'Poland',
        'United Kingdom',
        'United States',
      ]
    }
  }
}
</script>
