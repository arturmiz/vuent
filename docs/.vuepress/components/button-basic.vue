<template>
  <div class="playground">
    <vnt-button>Button</vnt-button>
  </div>
</template>

<script>
import { VntButton } from '../../../src/components';

export default {
  name: 'ButtonBasic',
  components: {
    VntButton
  }
}
</script>
