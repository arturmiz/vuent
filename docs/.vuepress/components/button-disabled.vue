<template>
  <div class="playground">
    <vnt-button disabled>Button</vnt-button>
  </div>
</template>

<script>
import { VntButton } from '../../../src/components';

export default {
  name: 'ButtonDisabled',
  components: {
    VntButton
  }
}
</script>
