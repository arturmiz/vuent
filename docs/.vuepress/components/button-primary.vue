<template>
  <div class="playground">
    <vnt-button primary>Primary</vnt-button>
  </div>
</template>

<script>
import { VntButton } from '../../../src/components';

export default {
  name: 'ButtonPrimary',
  components: {
    VntButton
  }
}
</script>
