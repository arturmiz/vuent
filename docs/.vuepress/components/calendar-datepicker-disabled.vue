<template>
  <div class="playground">
    <vnt-calendar-datepicker v-model="date" disabled />
    <div class="result">date: {{ date }}</div>
  </div>
</template>

<script>
import { VntCalendarDatepicker } from "../../../src/components";

export default {
  name: "CalendarDatepickerBasic",
  components: {
    VntCalendarDatepicker
  },
  data() {
    return {
      date: "2020-04-05"
    };
  }
};
</script>
