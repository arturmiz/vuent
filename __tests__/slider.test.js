import { createLocalVue, mount } from '@vue/test-utils';
import { VntSlider } from '@/components';
import { isInstalled } from './utils';

describe('Slider', () => {
  let wrapper;

  test('can be installed separately', () => {
    const localVue = createLocalVue();
    localVue.use(VntSlider);

    expect(isInstalled(localVue, VntSlider)).toBe(true);
  });

  describe('by default', () => {

    beforeAll(() => {
      wrapper = mount(VntSlider);
    });

    test('is not disabled', () => {
      expect(wrapper.vm.disabled).toBe(false);
    });

    test('label text is "Slider"', () => {
      expect(wrapper.vm.label).toBe('Slider');
    });

    test('min equals 0', () => {
      expect(wrapper.vm.min).toBe(0);
    });

    test('max equals 100', () => {
      expect(wrapper.vm.max).toBe(100);
    });

    test('value equals 50', () => {
      expect(wrapper.vm.value).toBe(50);
    });

    test('renders correctly', () => {
      expect(wrapper.html()).toMatchSnapshot();
    });

  });

  describe('can be disabled', () => {

    beforeAll(() => {
      wrapper = mount(VntSlider, {
        propsData: {
          disabled: true
        }
      });
    });

    test('is disabled', () => {
      expect(wrapper.vm.disabled).toBe(true);
    });

    test('renders correctly', () => {
      expect(wrapper.html()).toMatchSnapshot();
    });

  });

  describe('can have custom label', () => {

    beforeAll(() => {
      wrapper = mount(VntSlider, {
        propsData: {
          label: 'Custom Slider'
        }
      });
    });

    test('is correct', () => {
      expect(wrapper.vm.label).toBe('Custom Slider');
    });

    test('renders correctly', () => {
      expect(wrapper.html()).toMatchSnapshot();
    });

  });

  describe('can have custom min', () => {

    beforeAll(() => {
      wrapper = mount(VntSlider, {
        propsData: {
          min: 15
        }
      });
    });

    test('is correct', () => {
      expect(wrapper.vm.min).toBe(15);
    });

    test('renders correctly', () => {
      expect(wrapper.html()).toMatchSnapshot();
    });

  });

  describe('can have custom max', () => {

    beforeAll(() => {
      wrapper = mount(VntSlider, {
        propsData: {
          max: 200
        }
      });
    });

    test('is correct', () => {
      expect(wrapper.vm.max).toBe(200);
    });

    test('renders correctly', () => {
      expect(wrapper.html()).toMatchSnapshot();
    });

  });

  test('can have some other attribute set', () => {
    wrapper = mount(VntSlider, {
      attrs: {
        step: '5',
        'aria-label': 'a11y label'
      }
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  test('can have content set using slot', () => {
    wrapper = mount(VntSlider, {
      slots: {
        default: '<span>Custom content</span>'
      }
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  describe('when not disabled, updates passed value prop', () => {

    beforeAll(() => {
      wrapper = mount(VntSlider, {
        propsData: {
          value: 30
        }
      });

      const slider = wrapper.find('input');
      slider.element.value = 44;
      slider.trigger('input');
    });

    test('event is emitted', () => {
      expect(wrapper.emitted()['input']).toBeDefined();
    });

    test('new value state is emitted', () => {
      expect(wrapper.emitted()['input'][0]).toEqual([44]);
    });

  });

  describe('when disabled, doesn\'t change state', () => {

    beforeAll(() => {
      wrapper = mount(VntSlider, {
        propsData: {
          value: 3,
          disabled: true
        }
      });

      const slider = wrapper.find('input[type=range]');
      slider.element.value = 44;
      slider.trigger('input');
    });

    test('event is not emitted', () => {
      expect(wrapper.emitted()['input']).toBeUndefined();
    });

  });

});
