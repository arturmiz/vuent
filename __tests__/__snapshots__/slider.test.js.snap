// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Slider by default renders correctly 1`] = `
<div class="vnt-slider">
  <label class="vnt-slider__title">Slider</label>
  <input max="100" min="0" type="range" class="vnt-slider__input">
</div>
`;

exports[`Slider can be disabled renders correctly 1`] = `
<div class="vnt-slider">
  <label class="vnt-slider__title vnt-slider__title--disabled">Slider</label>
  <input disabled="disabled" max="100" min="0" type="range" class="vnt-slider__input">
</div>
`;

exports[`Slider can have content set using slot 1`] = `
<div class="vnt-slider">
  <label class="vnt-slider__title"><span>Custom content</span></label>
  <input max="100" min="0" type="range" class="vnt-slider__input">
</div>
`;

exports[`Slider can have custom label renders correctly 1`] = `
<div class="vnt-slider">
  <label class="vnt-slider__title">Custom Slider</label>
  <input max="100" min="0" type="range" class="vnt-slider__input">
</div>
`;

exports[`Slider can have custom max renders correctly 1`] = `
<div class="vnt-slider">
  <label class="vnt-slider__title">Slider</label>
  <input max="200" min="0" type="range" class="vnt-slider__input">
</div>
`;

exports[`Slider can have custom min renders correctly 1`] = `
<div class="vnt-slider">
  <label class="vnt-slider__title">Slider</label>
  <input max="100" min="15" type="range" class="vnt-slider__input">
</div>
`;

exports[`Slider can have some other attribute set 1`] = `
<div class="vnt-slider" step="5" aria-label="a11y label">
  <label class="vnt-slider__title">Slider</label>
  <input max="100" min="0" type="range" step="5" aria-label="a11y label" class="vnt-slider__input">
</div>
`;
